covid19-etl/etl/idph.py

```diff
@@ -66,169 +65,205 @@ def files_to_submissions(self):
-        print(f"Getting data for date: {today_str}")
+        print(
+            f"Latest submitted date: {latest_submitted_date}. Getting data for date: {today}"
+        )
 
-        # they changed the URL on April 1, 2020
-        if today > datetime.date(2020, 3, 31):
-            url = "http://www.dph.illinois.gov/sitefiles/COVIDTestResults.json"
-        else:
-            url = f"https://www.dph.illinois.gov/sites/default/files/COVID19/COVID19CountyResults{today_str}.json"
-        self.parse_file(latest_submitted_date, url)
+        self.parse_il_counties()
+
+        # convert date to datetime
+        latest_submitted_datetime = (
+            datetime.datetime(
+                latest_submitted_date.year,
+                latest_submitted_date.month,
+                latest_submitted_date.day,
+            )
+            if latest_submitted_date
+            else None
+        )
 
-    def parse_file(self, latest_submitted_date, url):
+        for county in self.county_dict:
+            self.parse_county_data(latest_submitted_datetime, county)
+
+        self.parse_state_data(latest_submitted_datetime)
+
+        print("Done in {} secs".format(int(time.time() - start)))
+
+    def parse_county_data(self, latest_submitted_date, county):
         """
         Converts a JSON files to data we can submit via Sheepdog. Stores the
         records to submit in `self.summary_locations` and `self.summary_clinicals`.
 
         Args:
-            latest_submitted_date (date): date for latest submitted date
-            url (str): URL at which the JSON file is available
+            latest_submitted_date (datetime): date for latest submitted date
+            county (str): county name
         """
-        print("Getting data from {}".format(url))
+        demographics = self.get_demographics(
+            # we need to start the day after `latest_submitted_date` but this is easier
+            start_date=latest_submitted_date,
+            end_date=datetime.date.today(),
+            county=county,
+        )
+
+        url = f"https://idph.illinois.gov/DPHPublicInformation/api/COVIDExport/GetCountyTestResultsTimeSeries?countyName={county}"
+        print("Getting county data from {}".format(url))
+
         with closing(self.get(url, stream=True)) as r:
-            data = r.json()
-            date = idph_get_date(data["LastUpdateDate"])
-
-            if latest_submitted_date and date == latest_submitted_date.strftime(
-                "%Y-%m-%d"
-            ):
-                print(
-                    "Nothing to submit: latest submitted date and date from data are the same."
+            daily_data = r.json()
+            for data in daily_data:
+                date = datetime.datetime.strptime(
+                    data["ReportDate"], "%Y-%m-%dT%H:%M:%S"
                 )
-                return
+                date_str = date.strftime("%Y-%m-%d")
+                if not latest_submitted_date or date <= latest_submitted_date:
```

Comment:
Some thing doesn't seem right here. We **shouldn't skip** if `latest_submitted_date is None` right?

Reply:
woops, should be fixed